python/pyspark/sql/datasource.py

```diff
@@ -298,6 +320,97 @@ def read(self, partition: InputPartition) -> Iterator[Union[Tuple, Row]]:
         ...
 
 
+class DataSourceStreamReader(ABC):
```

Comment:
Small hint: majority of methods we are introducing in Python streaming data source have corresponding methods in Scala API. You are OK to copy the same content, with modification to reflect the diff on contract if any.

Reply:
Great suggestion, I have copied corresponding method comments from scala API.